app/src/main/java/com/google/android/apps/youtube/app/settings/videoquality/VideoQualitySettingsActivity.java

```diff
@@ -0,0 +1,118 @@
+        if (dataString.equalsIgnoreCase("sponsorblock_settings")) {
+            trySetTitle(getIdentifier("sb_settings", "string"));
+            getFragmentManager()
+                    .beginTransaction()
+                    .replace(fragment, new SponsorBlockSettingsFragment())
+                    .commit();
+        } else if (dataString.equalsIgnoreCase("ryd_settings")) {
+            trySetTitle(getIdentifier("revanced_ryd_settings_title", "string"));
+            getFragmentManager()
+                    .beginTransaction()
+                    .replace(fragment, new ReturnYouTubeDislikeSettingsFragment())
+                    .commit();
+        } else {
+            trySetTitle(getIdentifier("revanced_settings", "string"));
+            getFragmentManager()
+                    .beginTransaction()
+                    .replace(fragment, new ReVancedSettingsFragment())
+                    .commit();
+        }
+    }
+
+    @Override
+    public void onDestroy() {
+        super.onDestroy();
+    }
+
+    public static ImageButton getImageButton(ViewGroup viewGroup) {
+        if (viewGroup == null)  return null;
+        int childCount = viewGroup.getChildCount();
+        for (int i = 0; i < childCount; i++) {
+            View childAt = viewGroup.getChildAt(i);
+            if (childAt instanceof ImageButton) {
+                return (ImageButton) childAt;
+            }
+        }
+        return null;
+    }
+
+    public static TextView getTextView(ViewGroup viewGroup) {
+        if (viewGroup == null) return null;
+        int childCount = viewGroup.getChildCount();
+        for (int i = 0; i < childCount; i++) {
+            View childAt = viewGroup.getChildAt(i);
+            if (childAt instanceof TextView) {
+                return (TextView) childAt;
+            }
+        }
+        return null;
+    }
+
+    private void trySetTitle(int i) {
+        try {
+            getTextView((ViewGroup) findViewById(getIdentifier("toolbar", "id"))).setText(i);
+        } catch (Exception e) {
+            LogHelper.printException(() -> ("Couldn't set Toolbar title"), e);
+        }
+    }
+
+    private static int getIdentifier(String name, String defType) {
+        Context appContext = ReVancedUtils.getContext();
+        assert appContext != null;
+        return appContext.getResources().getIdentifier(name, defType, appContext.getPackageName());
+    }
+
+    @SuppressLint("UseCompatLoadingForDrawables")
+    private void initImageButton() {
+        final var whiteArrow = "yt_outline_arrow_left_white_24";
+        final var darkArrow = "yt_outline_arrow_left_black_24";
+
+        final var arrow = ThemeHelper.isDarkTheme() ? whiteArrow : darkArrow;
+
+        try {
+            ImageButton imageButton = getImageButton((ViewGroup) findViewById(getIdentifier("toolbar", "id")));
+            imageButton.setOnClickListener(view -> VideoQualitySettingsActivity.this.onBackPressed());
```

Comment:
For this PR only this line of code is relevant, I do not know what the rest of the changes are for?

Reply:
Me neither. The only explanation he gave me can be found on the PR of the patches repo.

I will try to fix this PR following your requests, but I do not guarantee anything.